provider/aws/elbv2.go

```diff
@@ -0,0 +1,272 @@
+package aws
+
+import (
+	"encoding/json"
+	"errors"
+	"finala/config"
+	"finala/expression"
+	"finala/storage"
+	"finala/structs"
+	"regexp"
+	"strings"
+	"time"
+
+	awsClient "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/aws/aws-sdk-go/service/pricing"
+	log "github.com/sirupsen/logrus"
+)
+
+// ELBClientDescreptor is an interface defining the aws elbv2 client
+type ELBV2ClientDescreptor interface {
+	DescribeLoadBalancers(*elbv2.DescribeLoadBalancersInput) (*elbv2.DescribeLoadBalancersOutput, error)
+	DescribeTags(*elbv2.DescribeTagsInput) (*elbv2.DescribeTagsOutput, error)
+}
+
+// ELBManager describe TODO::appname ELB struct
+type ELBV2Manager struct {
+	client           ELBV2ClientDescreptor
+	storage          storage.Storage
+	cloudWatchCLient *CloudwatchManager
+	pricingClient    *PricingManager
+	metrics          []config.MetricConfig
+	region           string
+
+	namespace          string
+	servicePricingCode string
+}
+
+//
+type ARNManager struct {
+	Partition string
+	Service   string
+	Region    string
+	AccountID string
+	Resource  string
+}
+
+// DetectedELB define the detected AWS ELB instances
+type DetectedELBV2 struct {
+	Metric string
+	Region string
+	structs.BaseDetectedRaw
+}
+
+func ParseArn(arn string) (ARNManager, error) {
+	const (
+		arnDelimiter = ":"
+		arnSections  = 6
+		arnPrefix    = "arn:"
+
+		// zero-indexed
+		sectionPartition = 1
+		sectionService   = 2
+		sectionRegion    = 3
+		sectionAccountID = 4
+		sectionResource  = 5
+
+		// errors
+		invalidPrefix   = "arn: invalid prefix"
+		invalidSections = "arn: not enough sections"
+	)
+	if !strings.HasPrefix(arn, arnPrefix) {
+		return ARNManager{}, errors.New(invalidSections)
+	}
+	sections := strings.SplitN(arn, arnDelimiter, arnSections)
+	if len(sections) != arnSections {
+		return ARNManager{}, errors.New(invalidSections)
+	}
+
+	return ARNManager{
+		Partition: sections[sectionPartition],
+		Service:   sections[sectionService],
+		Region:    sections[sectionRegion],
+		AccountID: sections[sectionAccountID],
+		Resource:  sections[sectionResource],
+	}, nil
+
+}
+
+// TableName will set the table name to storage interface
+func (DetectedELBV2) TableName() string {
+	return "aws_elbv2"
+}
+
+// NewELBManager implements AWS GO SDK
+func NewELBV2Manager(client ELBV2ClientDescreptor, st storage.Storage, cloudWatchCLient *CloudwatchManager, pricing *PricingManager, metrics []config.MetricConfig, region string) *ELBV2Manager {
+
+	st.AutoMigrate(&DetectedELBV2{})
+
+	return &ELBV2Manager{
+		client:           client,
+		storage:          st,
+		cloudWatchCLient: cloudWatchCLient,
+		metrics:          metrics,
+		pricingClient:    pricing,
+		region:           region,
+
+		namespace:          "AWS/ApplicationELB",
+		servicePricingCode: "AmazonEC2",
+	}
+}
+
+// Detect check with ELB  instance is under utilization
```

Comment:
Please change the function description the ELBv2 and remove the double spaces

Reply:
Updated